src/bitcoin_adaptor.rs

```diff
@@ -45,14 +42,14 @@ impl BitcoinAdaptor {
     }
 
     /// Store some bitcoin object that has a bitcoin hash
-    pub fn put_hash_keyed<T>(&mut self, encodable: &T) -> Result<PRef, Box<dyn Error>>
-        where T: Serialize + BitcoinHash {
-        Ok(self.hammersbald.put_keyed(&encodable.bitcoin_hash()[..], serde_cbor::to_vec(encodable)?.as_slice())?)
+    pub fn put_hash_keyed<T>(&mut self, id: &sha256d::Hash, encodable: &T) -> Result<PRef, Box<dyn Error>>
+        where T: Serialize {
+        Ok(self.hammersbald.put_keyed(&id[..], serde_cbor::to_vec(encodable)?.as_slice())?)
     }
 
     /// Retrieve a bitcoin_object with its hash
     pub fn get_hash_keyed<T>(&self, id: &sha256d::Hash) -> Result<Option<(PRef, T)>, Box<dyn Error>>
-        where T: DeserializeOwned + BitcoinHash{
+        where T: DeserializeOwned {
```

Comment:
Why do we need the `BitcoinHash` removed here?

Reply:
See above, this will be removed in the next version

Reply:
@jrawsthorne I agree this makes the api easier to use. But I'm not as familiar with how the generic Into<> works. So I manually ran the tests (with `cargo test --features="bitcoin_support"`). The existing tests confirms the values passed in for the `id` argument is borrowed and can still be accessed after being passed into the get_hash_keyed function. So looks good to me.